packages/wasm/as/assembly/msgpack/ReadDecoder.ts

```diff
@@ -370,6 +370,14 @@ export class ReadDecoder extends Read {
     return false;
   }
 
+  isNextEnumValue(): bool {
+    // Enum integer value is in range 0 - 127
+    if (this.view.peekUint8() <= 127) {
```

Comment:
@dOrgJelli 
This is a bit sketchy and don't really know what to do with it.
Msgpack handles string and int that are under some limit without format prefix (e.g. 2 to msgpack is only 1 byte with 2) which then comes as variabe when peeking first Uint8 for format.
https://github.com/msgpack/msgpack-javascript/blob/e218974eac946276def928b8a1cb54f097db0a6e/src/Encoder.ts#L93 

This creates a problem for me as it is hard to know if it is a string or int from format.
Currently, I hardcoded that an enum has max 127 values and if format is less than 127 then it is an int value sent as enum argument, but would to hear your opinion on doing this some way smarter.

Reply:
Why not just use unsigned integers to represent enums? The size of the uint can be dynamically stored for compactness.

Reply:
See this PR to see how I solved this: https://github.com/Web3-API/monorepo/pull/152/commits/fe1ff1c525e55f2b38f16690cb5afa2e177d90e6